pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -309,27 +319,33 @@ func (s *Server) loadCACerts(caOpts *caOptions, dir string) error {
 // handleEvent handles the events on cacerts related files.
 // If create/write(modified) event occurs, then it verifies that
 // newly introduced cacerts are intermediate CA which is generated
-// from cuurent root-cert.pem. Then it updates and keycertbundle
+// from current root-cert.pem. Then it updates and keycertbundle
 // and generates new dns certs.
+// similarly, if crl file is created/modified,
+// it updates the crl file in keycertbundle and notifies the watcher
+// to replicate the new crl data.
 func handleEvent(s *Server) {
 	log.Info("Update Istiod cacerts")
 
 	var newCABundle []byte
 	var err error
+	var updateRootCA, updateCRL bool
 
-	currentCABundle := s.CA.GetCAKeyCertBundle().GetRootCertPem()
-
-	fileBundle, err := detectSigningCABundle()
+	fileBundle, err := detectSigningCABundleAndCRL()
 	if err != nil {
 		log.Errorf("unable to determine signing file format %v", err)
 		return
 	}
+
+	// check if CA bundle is updated
 	newCABundle, err = os.ReadFile(fileBundle.RootCertFile)
 	if err != nil {
 		log.Errorf("failed reading root-cert.pem: %v", err)
 		return
 	}
 
+	currentCABundle := s.CA.GetCAKeyCertBundle().GetRootCertPem()
+
 	// Only updating intermediate CA is supported now
```

Comment:
nit: not for you but this comment should be removed/changed since root CA rotation is now supported